src/routes/account/+page.server.ts

```diff
@@ -0,0 +1,107 @@
+import type { Actions, PageServerLoad } from './$types';
+import { error, redirect } from '@sveltejs/kit';
+import {
+  getStripeCustomerWithSubscriptions,
+  getBlendProSubscription,
+  getCustomerPortalSession,
+  createStripeSession,
+  isSubscribedToBlendPro,
+  isOrganizationMember,
+} from '$lib/server/subscriptionUtils';
+import { checkSessionAuth, getUserData, getUserOrganizations, isUserGlobalAdmin, readPath, writePath } from '$lib/server/firebaseUtils';
+
+let email = '';
+let name = '';
+let uid = '';
+
+export const load = (async ({ url, cookies }) => {
+  const actionParam = url.searchParams.get('action') ?? '';
+  uid = (await checkSessionAuth(cookies, {
+    loginRedirect: `account?${url.searchParams}`,
+    // authFunction: ({ uid: tokenUid }) => tokenUid === uid,
+  })).uid;
+
+  const userData = await getUserData(uid);
+  email = userData.email!;
+  name = userData.displayName ?? 'Blend User';
+  const isGlobalAdmin = isUserGlobalAdmin(uid); 
+  const organizations = getUserOrganizations(uid).then((orgIds) => {
+    return Promise.all(
+      (orgIds || []).map(async (orgId: string) => {
+        const organization = await readPath<Database.Organization>(`/organizations/${orgId}`);
+        return {  
+          id: orgId,
+          name: organization?.public.name,
+          role: (await isGlobalAdmin) ? 'admin' : organization?.private?.members?.[uid]?.role,
+        };
+      }),
+    );
+  });
```

Comment:
I know this was here before but looking now, I think this is cleaner (and should maintain strong types better) if we keep it all async/await
```suggestion
  const orgIds = await getUserOrganizations(uid);
  const organizations = await Promise.all(
    orgIds.map(async (orgId) => {
      const organization = await readPath<Database.Organization>(`/organizations/${orgId}`);
      return {
        id: orgId,
        name: organization?.public.name,
        role: (await isGlobalAdmin) ? 'admin' : organization?.private?.members?.[uid]?.role,
      };
    }),
  );
```